presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -136,7 +141,7 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         List<HostAddress> sortedCandidates = null;
         OptionalInt preferredNodeCount = OptionalInt.empty();
         for (Split split : splits) {
-            List<InternalNode> candidateNodes;
+            List<Pair<NodeStatus, InternalNode>> candidateNodes;
             switch (split.getNodeSelectionStrategy()) {
```

Comment:
out of the scope PR comment, we technically need not check the node selection strategy for every split. All the scheduled split in a task do have the same scheduling.